numpy/lib/npyio.py

```diff
@@ -248,6 +248,10 @@ def __getitem__(self, key):
         else:
             raise KeyError("%s is not a file in the archive" % key)
 
+    def __eq__(self, other):
+        if not isinstance(other, NpzFile):
+            return False
+        return dict(self.items()) == dict(other.items())
```

Comment:
In general, this is not going to work. Most of the values will be arrays, which will not evaluate to a `bool` when compared with `==` (as the `dict.__eq__()` implementations will try to do).

This will silence the LGTM alert, but does not actually address the underlying concern of the alert. What the alert is trying to say is that this subclass of `Mapping` adds attributes like `allow_pickle`, but didn't override `__eq__` to _include_ comparing the `allow_pickle` attribute and the others.

All said, I don't think we care much about `__eq__` on this class. It's not really useful for anything. We can just ignore the LGTM alert for this.

Reply:
Would it be fine if I undo that commit and silence the alert under this PR?

Reply:
In general, I think we just ignore them. [We may just turn it off](https://www.mail-archive.com/numpy-discussion@python.org/msg05376.html). I don't see any other LGTM-suppression comments in our codebase, so I would not start now.

Reply:
You don't actually have to compare the values though, as a filename (_i.e._ key) always has a unique array attached to it (barring things like symbolic links). So I think we could get away with just comparing the `keys()` of both mappings, plus some attributes of interest:
``` python
def __eq__(self, other):
    if not isinstance(other, NpzFile):
        return NotImplemented
    return (
        self.pickle_kwargs == other.pickle_kwargs and 
        self.allow_pickle == other.allow_pickle and 
        self.keys() == other.keys()
    )
```

Reply:
I think a LGTM alert is not a good enough reason to add this method. Is there any real need for this?